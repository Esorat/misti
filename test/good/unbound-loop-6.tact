struct S {
    a: Int;
}

contract Test {
    receive("test") {
        let s: S = S {
            a: 1,
        };
        let i: Int = 0;
        while (i < s.a) { // no false positive
            i += 1;
        }
    }
}
