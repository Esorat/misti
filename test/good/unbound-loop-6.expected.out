[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop-6.tact:7:13:
  6 |     receive("test") {
> 7 |         let s: S = S {
                  ^
  8 |             a: 1,
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables